devtools/create_project/xcode.cpp

```diff
@@ -54,39 +62,172 @@ namespace CreateProjectTool {
+	std::string n, ext;
+	splitFilename(fileName, n, ext);
+
+	// Note that the difference between this and the general producesObjectFile is that
+	// this one adds Objective-C(++), and removes asm-support.
+	if (ext == "cpp" || ext == "c" || ext == "m" || ext == "mm")
+		return true;
+	else
+		return false;
+}
+
+XCodeProvider::Group::Group(XCodeProvider *objectParent, const std::string &groupName, const std::string &uniqueName, const std::string &path) : Object(objectParent, uniqueName, groupName, "PBXGroup", "", groupName) {
+	addProperty("name", name, "", SettingsNoValue|SettingsQuoteVariable);
+	addProperty("sourceTree", "<group>", "", SettingsNoValue|SettingsQuoteVariable);
+	
+	if (path != "") {
+		addProperty("path", path, "", SettingsNoValue|SettingsQuoteVariable);
+	}
+	_childOrder = 0;
+	_treeName = uniqueName;
+}
+
+void XCodeProvider::Group::ensureChildExists(const std::string &name) {
+	std::map<std::string, Group*>::iterator it = _childGroups.find(name);
+	if (it == _childGroups.end()) {
+		Group *child = new Group(parent, name, this->_treeName + '/' + name, name);
+		_childGroups[name] = child;
+		addChildGroup(child);
+		parent->_groups.add(child);
+	}
+}
+
+void XCodeProvider::Group::addChildInternal(const std::string &id, const std::string &comment) {
+	if (properties.find("children") == properties.end()) {
+		Property children;
+		children.hasOrder = true;
+		children.flags = SettingsAsList;
+		properties["children"] = children;
+	}
+	properties["children"].settings[id] = Setting("", comment + " in Sources", SettingsNoValue, 0, _childOrder++);
+	if (_childOrder == 1) {
+		// Force children to use () even when there is only 1 child.
+		// Also this enforces the use of "," after the single item, instead of ; (see writeProperty)
+		properties["children"].flags |= SettingsSingleItem;
+	} else {
+		properties["children"].flags ^= SettingsSingleItem;
+	}
+
+}
+
+void XCodeProvider::Group::addChildGroup(const Group* group) {
+	addChildInternal(parent->getHash(group->_treeName), group->_treeName);
+}
+
+void XCodeProvider::Group::addChildFile(const std::string &name) {
+	std::string id = "FileReference_" + _treeName + "/" + name;
+	addChildInternal(parent->getHash(id), name);
+	FileProperty property = FileProperty(name, name, name, "\"<group>\"");
+
+	parent->addFileReference(id, name, property);
+	if (producesObjectFileOnOSX(name)) {
+		parent->addBuildFile(_treeName + "/" + name, name, parent->getHash(id), name + " in Sources");
+	}
+}
+
+void XCodeProvider::Group::addChildByHash(const std::string &hash, const std::string &name) {
+	addChildInternal(hash, name);
+}
+
+XCodeProvider::Group *XCodeProvider::Group::getChildGroup(const std::string &name) {
+	std::map<std::string, Group*>::iterator it = _childGroups.find(name);
+	assert(it != _childGroups.end());
+	return it->second;
+}
+
+XCodeProvider::Group *XCodeProvider::touchGroupsForPath(const std::string &path) {
+	if (_rootSourceGroup == nullptr) {
```

Comment:
I had to change the nullptr to NULL to get create_project to compile with the default options (C99). Since we don't really use C11 functionality for create_project, I don't believe there's any advantage of using nullptr here